test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java

```diff
@@ -457,6 +462,213 @@ public void checkpointWriterInterleavedTest() throws Exception {
+        Token snapshot = TransactionalContext
+                .getCurrentContext()
+                .getSnapshotTimestamp();
+        try {
+            cpw.startCheckpoint(snapshot);
+            cpw.appendObjectState(m.entryStream());
+            cpw.finishCheckpoint();
+
+            // Instantiate new runtime & map.
+            setRuntime();
+            Map<String, String> m2 = instantiateStringMap(streamName);
+            for (int i = 0; i < numKeys; i++) {
+                assertThat(m2.get(keyPrefix + i)).describedAs("get " + i)
+                        .isEqualTo(m.get(keyPrefix + i) + mutationSuffix);
+            }
+        } finally {
+            r.getObjectsView().TXEnd();
+        }
+    }
+
+    @Test
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void checkpointWriterSizeLimitMarginTest() throws Exception {
+        final String streamName = "mystream6";
+        final UUID streamId = CorfuRuntime.getStreamID(streamName);
+        final String keyPrefix = "a-prefix";
+        final String author = "Me, myself, and I";
+        // max write size set to 25 MB
+        final int maxWriteSize = 25 << 20;
+        final double marginRatio = 0.02;
+
+        StreamingMap<String, String> m = instantiateStringMap(streamName);
+
+        // set the max write size to 25 MB
+        getRuntime().getParameters().setMaxWriteSize(maxWriteSize);
+        CheckpointWriter cpw = new CheckpointWriter(getRuntime(), streamId, author, (CorfuTable) m);
+        cpw.setSerializer(serializer);
+        // use default batch size which is 50
+
+        int i = 0;
+        int numBytesPerCheckpointEntry = 0;
+        // each entry is 1 MB
+        String payload = getRandomStringOfSize(1 << 20);
+        int nextSmrEntrySize = getSerializedSMREntrySize(keyPrefix + i, payload,
+                cpw.getKeyMutator(), cpw.getValueMutator(), cpw.getSerializer());
+
+        while (numBytesPerCheckpointEntry + nextSmrEntrySize < (1 - marginRatio) * getRuntime().getParameters().getMaxWriteSize()) {
+            m.put(keyPrefix + i, payload);
+            payload = getRandomStringOfSize(1 << 20);
+            numBytesPerCheckpointEntry += nextSmrEntrySize;
+            nextSmrEntrySize = getSerializedSMREntrySize(keyPrefix + i, payload,
+                    cpw.getKeyMutator(), cpw.getValueMutator(), cpw.getSerializer());
+            System.out.println(String.format("After putting entry %s, numBytesPerCheckpointEntry is: %s",
+                    (keyPrefix + i), numBytesPerCheckpointEntry));
+            System.out.println("nextSmrEntrySize: " + nextSmrEntrySize);
+            System.out.println("---------------------------------");
+            i++;
+        }
+
+        System.out.println("maxWriteSize: " + getRuntime().getParameters().getMaxWriteSize());
+        System.out.println("Total numBytesPerCheckpointEntry is: " + numBytesPerCheckpointEntry);
+        System.out.println("(1 - marginRatio) * maxWriteSize is: " +
```

Comment:
I don't think we use `System.out.println` in testing either. Use a logger, and if you are not want to do debugging, modify the appropriate XML file (I think it's `logback-classic.xml`)

Reply:
We actually use `System.out.println` in other unit tests here (`consumeRawStreamCheckpointTrim` and `getCheckpointId`).